Mathlib/Tactic/ToAdditive.lean

```diff
@@ -435,6 +435,44 @@ def isInternal' (declName : Name) : Bool :=
   | .str _ s => "match_".isPrefixOf s || "proof_".isPrefixOf s || "eq_".isPrefixOf s
   | _        => true
 
+/-- Find the target name of `pre` and all created auxiliary declarations. -/
+def findTargetName (env : Environment) (src pre tgt_pre : Name) : CoreM Name :=
+  let privatePrefix := privateHeader ++ env.mainModule |>.num 0
+  /- This covers auxiliary declarations like `match_i` and `proof_i`. -/
+  if let some post := pre.isPrefixOf? src then
+    return tgt_pre ++ post
+  /- This covers equation lemmas (for other declarations). -/
+  else if let some post := privatePrefix.isPrefixOf? src then
+    match findTranslation? env post.getPrefix with
+    -- this is an equation lemma for a declaration without `to_additive`. We will skip this.
+    | none => return src
+    -- this is an equation lemma for a declaration with `to_additive`. We will additivize this.
+    -- Note to future self: if this errors we could do this instead by calling `getEqnsFor?`
+    | some addName => return src.updatePrefix <| privatePrefix ++ addName
+  -- Note: this additivizes lemmas generated by `simp`.
+  -- Todo: we do not currently check whether such lemmas actually should be additivized.
+  else if let some post := env.mainModule ++ `_auxLemma |>.isPrefixOf? src then
+    return env.mainModule ++ `_auxAddLemma ++ post
+  else
+    throwError "internal @[to_additive] error."
+
+/-- Returns a `NameSet` of all auxiliary constants in `e` that might have been generated
+when adding `pre` to the environment.
+Examples include `pre.match_5`, `Mathlib.MyFile._auxLemma.3` and
+`_private.Mathlib.MyFile.someOtherNamespace.someOtherDeclaration._eq_2`.
+The last two examples may or may not have been generated by this declaration.
+The last example may or may not be the equation lemma of a declaration with the `@[to_additive]`
+attribute. We will only translate it has the `@[to_additive]` attribute.
+-/
+def findAuxDecls (e : Expr) (pre mainModule : Name) : NameSet :=
+let auxLemma := mainModule ++ `_auxLemma
+let privatePrefix := privateHeader ++ mainModule |>.num 0
+e.foldConsts ∅ fun n l ↦
```

Comment:
If you only want to get declarations from the current file, then iterate over `e.constants.map₂`.

Reply:
I think you misinterpret `e` as an environment instead of an expression. If not, I don't understand your comment.

Reply:
Indeed I did.